framework/src/st_things/things_stack/framework/things_data_manager.c

```diff
@@ -345,9 +345,33 @@ static struct st_resource_type_s *create_resource_type()
 
 static void delete_resource_type(st_resource_type_s *type)
 {
+	if (type == NULL) {
```

Comment:
please check coding rule 
tab size is 4.

Reply:
Tab size is aligning with the existing code. Please re-check.